### PR TITLE
Change track selector on mentoring queue

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-project-management/issues/169.

### CODE REVIEW

File: app/javascript/components/mentoring/Queue.tsx [hunk 1]
```diff
@@ -136,6 +130,8 @@ export const Queue = ({
           isFetching={isTrackListFetching}
           value={selectedTrack}
           setValue={setSelectedTrack}
+          cacheKey={TRACKS_LIST_CACHE_KEY}
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/Queue.tsx:126]: Does this work correctly if you login, view this page (cache is created), logout, login as a different user and visit this page again? In other words: is this tied to the current user?

File: app/javascript/components/mentoring/queue/TrackFilterList.tsx [hunk 2]
```diff
@@ -55,39 +60,88 @@ type Props = {
...
-              key={track.id}
-              onChange={(e) => handleChange(e, track)}
-              checked={value?.id === track.id}
-              {...track}
-            />
-          ))}
+    <React.Fragment>
+      {isFetching ? <span>Fetching...</span> : null}
+      <button
+        aria-label="Button to open the track filter"
```
Comment (ErikSchierboom) [inline app/javascript/components/mentoring/queue/TrackFilterList.tsx]: Is the "Button to" prefix needed? It looks like a screenreader could recognize that it is a button from the `<button>` element itself. But I don't know anything about accessibility to be honest so 🤷

**Review by ErikSchierboom — APPROVED** [review]

**Comment by iHiD** [comment]
@kntsoriano This is nearly there. It's looking great now - nice work! 🙂 

Three TODOs
- [x] Changing track in the selector should close the dropdown (in either the Queue or the Inbox)
- [x] The TODOs in this PR
- [x] Fix the tests.
- [x] Check [Erik's comment](https://github.com/exercism/website/pull/683#discussion_r605575537)

File: app/javascript/components/mentoring/Queue.tsx (hunk 1 above)
Comment (kntsoriano) [reply app/javascript/components/mentoring/Queue.tsx:126]: It is not scoped per user. if the whole process (logging in and out) happens in less than 5 minutes, the cached values will be used.

File: app/javascript/components/mentoring/queue/TrackFilterList.tsx (hunk 2 above)
Comment (kntsoriano) [reply app/javascript/components/mentoring/queue/TrackFilterList.tsx]: Agreed :) I will remove it.

**Comment by iHiD** [comment]
Tests still fail, @kntsoriano

**Pull request merged**